torch/_inductor/ir.py

```diff
@@ -7392,7 +7392,7 @@ def __init__(
         kwargs=None,
     ):
         super().__init__(
-            NoneLayout(layout.device),
+            layout,
```

Comment:
why did we need this change ?

Reply:
I found that if layout is None, then during codegen the program doesn't flush correctly [here](https://fburl.com/code/0vuhx3hp), and then the generated python code will be incorrect.